### PR TITLE
Use Bootstrap for MCQ and MSQ other option textbox #4014

### LINKED ISSUE #4014: Use Bootstrap for MCQ and MSQ other option textbox

### PR DESCRIPTION
**Pull request by JosephineKwa**
Fixes #4014

MCQ options:
mcq
MSQ options:
msq

### CODE REVIEW

**Comment by kanghj** [comment]
It seems quite bad that this isn't tested? Let's add a test?

**Comment by YongJieYongJie** [comment]
Maybe we can display the "Other" option without the textbox first.
 - When user click on it, we add a textbox (or textarea) with the words "If other, please specify:" at the bottom of the question
 - For MSQ, we should also hide the "none of the above" when user clicks on the "Other" option

A brief Google search reveal these possible layouts:
 - "Other" option separated from the other options
image

- "Other" option in the same list as the other options
image

**Comment by JosephineKwa** [comment]
Would this be a better alternative:
other1
other2

For point 1, I would prefer to display the "Other" option in the same list as the other options instead of showing a text box only upon clicking as the latter would add additional javascript to an already straightforward layout.

For point 2, currently checking the "None of the above" option will uncheck the other options to ensure that there won't be a contradictory response.

If we hide the "None of the above" option when the "Other" option is selected, the user would have to uncheck "Other" before the "None of the above" option would reappear. Again, I feel that this might be adding unnecessary javascript :stuck_out_tongue: .

**Comment by YongJieYongJie** [comment]
I think this works too, maybe there should be a bigger gap between the "other" and the box.

Also, do you think it'll make sense if "other" is below "none of the above"?

**Comment by JosephineKwa** [comment]
Hm, it might look better as doing so would get rid of the space between the "Other" and "None of the above" checkboxes. However I would prefer to have the "None of the above" option below the "Other" option as selecting this would have the meaning of excluding the "Other" option as part being of the "above".

**Comment by JosephineKwa** [comment]
Added some spacing the other label and the textbox
other3
other4

File: src/test/java/teammates/test/pageobjects/AppPage.java [hunk 1]
```diff
@@ -224,6 +224,11 @@ public void waitForElementToBeClickable(WebElement element) {
         WebDriverWait wait = new WebDriverWait(browser.driver, TestProperties.TEST_TIMEOUT);
         wait.until(ExpectedConditions.elementToBeClickable(element));
     }
+    
+    public void waitForElementToBeClickable(String elementId) {
+        WebDriverWait wait = new WebDriverWait(browser.driver, TestProperties.TEST_TIMEOUT);
+        wait.until(ExpectedConditions.elementToBeClickable(browser.driver.findElement(By.id(elementId))));
+    }
```
Comment (wkurniawan07) [inline src/test/java/teammates/test/pageobjects/AppPage.java]: Let's not duplicate the method: just call the pre-defined `waitForElementToBeClickable`. At the very least make use of the previous method here.

File: src/test/java/teammates/test/pageobjects/AppPage.java [hunk 2]
```diff
@@ -722,6 +727,10 @@ public boolean isElementCovered(WebElement element) {
     public void verifyUnclickable(WebElement element) {
         assertNotNull(element.getAttribute("disabled"));
     }
+    
+    public void verifyUnclickable(String elementId) {
+        assertNotNull(browser.driver.findElement(By.id(elementId)).getAttribute("disabled"));
```
Comment (wkurniawan07) [inline src/test/java/teammates/test/pageobjects/AppPage.java]: Similarly here.

File: src/test/resources/pages/studentFeedbackSubmitPagePartiallyFilled.html [hunk 3]
```diff
@@ -583,10 +583,10 @@
               <label for="input">
                 To:
               </label>
-              <select class="participantSelect middlealign newResponse form-control" name="responserecipient-6-0" style="display:none;max-width:125px">
-                <option selected="" value="">
+              <select class="participantSelect middlealign form-control" name="responserecipient-6-0" style="display:none;max-width:125px">
+                <option value="">
                 </option>
-                <option value="SFSubmitUiT.alice.b@gmail.tmt">
+                <option selected="" value="SFSubmitUiT.alice.b@gmail.tmt">
```
Comment (wkurniawan07) [inline src/test/resources/pages/studentFeedbackSubmitPagePartiallyFilled.html]: Any idea why this change is needed?

Comment (JosephineKwa) [inline src/test/resources/pages/studentFeedbackSubmitPagePartiallyFilled.html]: Originally, at this point in the test, this question wouldn't have been answered. Hence it would not have an existing response, and the first option would have the selected attribute(line 183 of `FeedbackSubmissionEditPageData`). 

However due to line 161-173 of `StudentFeedbackSubmitPageUiTest`, which adds a response for question 6 and question 8, this question (qn 6) will already have been answered at this point. 

Hence, line 186 of `FeedbackSubmissionEditPageData` will result in `SFSubmitUiT.alice.b@gmail.tmt` having the selected attribute.
